src/inmanta/ast/type.py

```diff
@@ -614,6 +612,11 @@ def validate(self, value: object) -> bool:
     def type_string_internal(self) -> str:
         return "Union[%s]" % ",".join(t.type_string_internal() for t in self.types)
 
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, Union):
+            return NotImplemented
+        return self.types == other.types
```

Comment:
This only covers the case where the types are defined in the same order. And we don't flatten nested unions. Then again, sorting them is probably non-trivial, let alone the flattening.

I have to check what this is used for again to see if this is acceptable.

Reply:
And `NullableType` vs `Union` with null. But that might just be something we have to clean up sometime, rather than hack around now.

Reply:
@wouterdb I have the impressino that the `__eq__` implementation isn't all that important, and mainly used for testing. Can you confirm?

Reply:
That is correct